vm/interpreter/src/default_runtime.rs

```diff
@@ -1135,37 +1142,85 @@ fn transfer<BS: BlockStore>(
     from: &Address,
     to: &Address,
     value: &TokenAmount,
+    version: NetworkVersion,
 ) -> Result<(), ActorError> {
-    if from == to {
-        return Ok(());
-    }
+    let (to_id, mut f) = if version >= NetworkVersion::V15 {
+        if value.is_negative() {
```

Comment:
Out of curiosity, sending a negative value is an error, sending to same address is noop - what about sending 0? I understand that Filecoin allows such transactions? If so, do you know what's the purpose of them?

Reply:
I don't think you can create a working transaction with 0 FIL because you will have to pay gas at some point (there's no free lunch).

Reply:
Makes sense. I guess there won't be that many `0 FIL` transactions so potential performance gain from short-circuiting on that would be negligible. Looks good to me, thanks!

Reply:
There are some check for `!= 0` transfer but this is done after charging.

See [this line](https://github.com/ChainSafe/forest/blob/elmattic/actors-review-F26/vm/interpreter/src/default_runtime.rs#L430).